### ISSUE TITLE
Package for Debian/Ubuntu

### ISSUE THREAD

**Issue by xamanu** [organicmaps/organicmaps#395]
There is no good FOSS offline map app for the Linux desktop. Organic Maps contains instructions on how to build this app for the desktop. Wondering if it would make sense to package Organic Apps for Debian and make it available  for this Operating System and its derivates such as Ubuntu, etc. It seems this would fill a gap and it also makes sense the more hardware like Pinephones are around that build on more regular Linux systems.

**Comment by xamanu** [comment]
@rtsisyk what would be your thoughts about this?

**Comment by rtsisyk** [comment]
Hi @xamanu, thank you for your interest to this project!

We are not focusing on the desktop version, but if somebody can contribute and maintain the Debian package it would be great. There was [another request](https://github.com/organicmaps/organicmaps/discussions/375) for a desktop version, please collaborate with @KAMiKAZOW.

**Comment by xamanu** [comment]
Sure, I'm going to build and test it, if it runs nicely on a Linux system, I'd consider helping out with the packaging.

**Comment by KAMiKAZOW** [comment]
> please collaborate with @KAMiKAZOW.

Huh? Where did you get the impression that I have any interest in Ubuntu?

**Comment by xamanu** [comment]
For reference, the related [Debian "bug"](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=986232).

**Comment by xamanu** [comment]
Currently, the software is not working perfectly on a Linux system, yet. And data download is being done via a private URL.

For me the two big questions would be:

* are maintainers interested (at some time - no rush; no pressure!) to support actively a desktop build, in terms of fixing bugs, etc.?
* how do maintainers envision data consumption of the desktop program (currently, the route to the data download, is not part of the source code and hold private)?

**Comment by rtsisyk** [comment]
See https://github.com/organicmaps/organicmaps/pull/398

**Comment by fgaz** [comment]
> Thanks for the great work on providing a truly free and respectful OpenStreetMap based vector map app!

seconding this!

I just packaged this software for [NixOS](https://nixos.org), so here's some more info for other packagers & feedback for the project:

* I had to disable `tools/unix/check_cert.sh` since it's dependent on time and affects reproducible builds
* I applied #418 
* I see that lots of dependencies are vendored. It'd be nice to be able to link to the system ones, even just because of the build times (icu in particular takes a lot)
* I had to manually install the OMaps binary, the data directory, and the icon and desktop files, in addition to running the normal ninja/cmake install.
  * it'd be nice if install commands for those items were added in the cmake files. I'd make a patch myself, but I'm not very familiar with cmake.
* I removed the xmlfw executable that was automatically installed, since it looks like OMaps does not need it (it only uses the library) and it may conflict with the system's one
* I wrapped the OMaps executable with a script that adds these flags:
  * `-resources_path $PREFIX/share/organicmaps/data` (where I installed `data`)
  * `-data_path "''${XDG_DATA_HOME:-''${HOME}/.local/share}/OMaps"` this has to be separate, since resources_path is not writable
* the script also creates the data_path if not existing before starting the program, since it's not automatically created

The package is pending review, you can see its source at https://github.com/NixOS/nixpkgs/pull/123259

**Comment by rtsisyk** [comment]
>I had to disable tools/unix/check_cert.sh since it's dependent on time and affects reproducible builds

I have no idea about this script. 
CC: @biodranik 

>I applied #418

Merged. Thanks for PR!

>I see that lots of dependencies are vendored. It'd be nice to be able to link to the system ones, even just because of the build times (icu in particular takes a lot)

Please feel free to send a patch to our CMake scripts to allow system dependencies.

> I had to manually install the OMaps binary, the data directory, and the icon and desktop files, in addition to running the normal ninja/make install. 

CMake can be used to install all required files, please make a patch.

>I removed the xmlfw executable that was automatically installed, since it looks like OMaps does not need it (it only uses the library) and it may conflict with the system's one

Make sense. 

> wrapped the OMaps executable with a script that adds these flags: 
>the script also creates the data_path if not existing before starting the program, since it's not automatically created

I think that we can try to patch default paths in the executable.

> The package is pending review, you can see its source at NixOS/nixpkgs#123259

Thanks for your work on Nix package. We appreciate that.  You are welcome to send all required patches directly to our upstream instead of applying them in packages.

**Comment by biodranik** [comment]
> I had to disable tools/unix/check_cert.sh since it's dependent on time and affects reproducible builds

It looks like embedding some client certificates to avoid malicious connections to rogue servers. Maps.me used it for user passport auth. We don't use these certs at the moment but may enable them in the future if cloud backups will be introduced.

**Comment by andrewshadura** [comment]
I was going to work on a package, but it’s going to be a long road because of a lot of vendored libraries.

Probably a Flatpak package might get usable results quicker.

**Comment by fgaz** [comment]
Same for the NixOS package :-/

The package is working and the PR is there, but we're waiting to merge until we can use system libraries, the biggest and most security sensitive ones at least.

I'd do this myself, but as I wrote above I'm not knowledgeable about cmake, so if anyone could help with this that'd be great! :pray:

**Comment by suoko** [comment]
Did anybody try to build a package for xenial? That way Ubuntu touch users can test it using the libertine container.

**Comment by fgaz** [comment]
> I was going to work on a package, but it’s going to be a long road because of a lot of vendored libraries.

Linking with system libraries was just enabled for freetype/icu!

#1724 #1877 #1870 

Maybe I could use that PR as a base to try to do the same with other libraries

**Comment by fgaz** [comment]
https://github.com/NixOS/nixpkgs/pull/123259 is in, we have the first distro package!

**Comment by andrewshadura** [comment]
@fgaz, thanks for your work on devendoring 🙂

**Comment by fgaz** [comment]
Thanks but that wasn't me, see #1877. And unfortunately some other libraries are still bundled, so the package is not perfect

**Comment by taivlam** [comment]
The `organicmaps` [package](https://salsa.debian.org/debian/organicmaps) is on Salsa (Debian's GitLab), but it's not yet in Debian's package [tracker](https://tracker.debian.org/).

**Comment by andrewshadura** [comment]
Yes, it’s still [in review](https://ftp-master.debian.org/new/organicmaps_2024.05.11.3+ds2-2.html).
